mllib/src/main/scala/org/apache/spark/ml/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -115,7 +122,10 @@ private[spark] object DecisionTreeMetadata extends Logging {
     }
     require(numFeatures > 0, s"DecisionTree requires number of features > 0, " +
       s"but was given an empty features vector")
-    val numExamples = input.count()
+    val (numExamples, weightSum) = input.aggregate((0L, 0.0))(
```

Comment:
Nit: use a `case` expression to avoid the `._1`, `._2`. It'll be more readable

Reply:
good idea, done!